Issue 4528 - Fix cn=monitor SCOPE_ONE search

droideck opened the pull request:
Bug Description: While doing a ldapsearch on "cn=monitor" is throwing
err=32 with -s one.

Fix Description: 'cn=monitor' is not a real entry so we should not
trying to check if the searched suffix (cm=monitor or its children)
belongs to the searched backend.

Fixes: #4528

Reviewed by: ?

mreynolds389 commented on ldap/servers/slapd/opshared.c:
```diff
@@ -809,6 +812,7 @@ op_shared_search(Slapi_PBlock *pb, int send_result)
                     }
                 }
             }
+            slapi_sdn_free(&monitorsdn);
```
You also need to free this before we "goto next_be" at line 794

Firstyear commented on ldap/servers/slapd/opshared.c:
```diff
@@ -764,8 +765,10 @@ op_shared_search(Slapi_PBlock *pb, int send_result)
+             * -> don't change the search parameters in pblock
+             * Also, we skip this block for 'cn=monitor' search and its subsearches
+             * as they are done by callbacks from monitor.c */
+            if ((be_suffix != NULL) && !slapi_sdn_issuffix(basesdn, monitorsdn)) {
```
This logic may be clearer as:

```
if (!( be_suffix == NULL) || slapi_sdn_issuffix() ))
```

droideck replied:
Probably just a typo, but I think you meant  (!((be_suffix == NULL) || slapi_sdn_issuffix())) - the third parenthesis. 
But yeah, I agree, it sounds more cleaner:) Fixed

Firstyear reviewed (changes requested):
Does slapi_sdn_done try to free the pointer? because this is on the stack not the heap it may be trying to free it incorrectly, which could cause a crash/failure.

droideck committed:
Issue 4528 - Fix cn=monitor SCOPE_ONE search

Bug Description: While doing a ldapsearch on "cn=monitor" is
throwing err=32 with -s one.

Fix Description: 'cn=monitor' is not a real entry so we should not
trying to check if the searched suffix (cm=monitor or its children)
belongs to the searched backend.

Fixes: #4528

Reviewed by: ?
